scripts/post-install/post-install.js

```diff
@@ -0,0 +1,40 @@
+const fs = require('fs');
+const Logger = require('../utils/logger');
+
+/**
+ * Validates the installation of terra-toolkit.
+ * Logs a warning if terra-toolkit is detected as a hard dependency.
+ * terra-toolkit is expected to always be installed as a devDependency.
+ */
+function validateInstall() {
+  /**
+   * INIT_CWD returns the initial working directory of the npm install.
+   * See the following issue for more information: https://github.com/npm/npm/issues/16990
+   * Note: This feature was introduced in npm 5.4.0.
+   */
+  if (!process.env.INIT_CWD) {
+    Logger.warn('WARNING: [terra-toolkit] - INIT_CWD was undefined. This is likely due to an outdated version of npm. Please consider upgrading.');
```

Comment:
I would like to add toolkit 
```suggestion
    Logger.warn('WARNING: [terra-toolkit] - INIT_CWD was undefined. This is likely due to an outdated version of npm. Please consider upgrading; terra-toolkit does not support this npm version.');
```

Reply:
I don't know if that description is entirely accurate, but it is more persuasive to upgrade. I appended the additional message: https://github.com/cerner/terra-toolkit/pull/329/commits/9ee6b3d77128f8aeca16e74417b45a53ea3399c9

Reply:
didn't you say the node version determined the npm version? or did I misunderstand? We define the minimum supported version as [node 8.9.2](https://github.com/cerner/terra-toolkit/blob/master/package.json#L7)

Reply:
When you use nvm it installs an npm version with the node version, but I think you can technically set npm to be any version of your choosing

Reply:
ahh

Reply:
Removed the unsupported message after the discussion here: https://github.com/cerner/terra-toolkit/pull/329#discussion_r322837728

Reply:
I still approve 😄 its a pretty blurry line